src/main/java/org/assertj/core/internal/Maps.java

```diff
@@ -732,23 +732,26 @@ public void assertHasSameSizeAs(AssertionInfo info, Map<?, ?> map, Map<?, ?> oth
    * @param entries the entries that should be in the actual map.
    * @throws AssertionError if the actual map is {@code null}.
    * @throws NullPointerException if the given entries array is {@code null}.
-   * @throws IllegalArgumentException if the given entries array is empty.
    * @throws AssertionError if the actual map does not contain the given entries, i.e. the actual map contains some or
-   *           none of the given entries, or the actual map contains more entries than the given ones.
+   *           none of the given entries, or the actual map contains more entries than the given ones, or if entries is
+   *           empty.
    */
   public <K, V> void assertContainsOnly(AssertionInfo info, Map<K, V> actual,
                                         @SuppressWarnings("unchecked") Map.Entry<? extends K, ? extends V>... entries) {
     doCommonContainsCheck(info, actual, entries);
     if (actual.isEmpty() && entries.length == 0) {
       return;
     }
-    failIfEmpty(entries);
 
     Set<Map.Entry<? extends K, ? extends V>> notFound = new LinkedHashSet<>();
     Set<Map.Entry<? extends K, ? extends V>> notExpected = new LinkedHashSet<>();
 
     compareActualMapAndExpectedEntries(actual, entries, notExpected, notFound);
 
+    if(!actual.isEmpty() && entries.length == 0) {
+      throw failures.failure(info, shouldContainOnly(actual, entries, notFound, notExpected));
+    }
+
```

Comment:
please do the same for `assertContainsExactly` and move the code in a private method to avoid code duplication

Reply:
you don't need to check that actual is not empty because if entries is empty, actual can't due to the check line 742

Reply:
change the failure to use `ShouldBeEmpty.shouldBeEmpty` and move the `if` before creating `notFound`

Reply:
> please do the same for `assertContainsExactly` and move the code in a private method to avoid code duplication

In assertContainsExactly there is already a check to validate that the size of both actual and entries is same, which will take care of scenario where actual is not empty but entries is empty.  So doing the same thing again is of no significance. WDYT?
I have updated the test Maps_assertContainsExactly_Test. should_fail_if_given_entries_array_is_empty to test this scenario.